### DISCUSSION TITLE
How to redirect user to origin url?

### DISCUSSION THREAD

**Discussion by nicitaacom** [supabase/supabase#18751] [Questions] [answered]
I want if I in development redirect user to http://localhost:3000/
If I in production redirect user to https://23-store.vercel.app/

How can I do it?

![image](https://github.com/supabase/supabase/assets/39565703/20ce791a-244e-4f07-bd5b-d217c559a6f5)

Now in production I get redirect to localhost:3000

I use location.origin

18 line - https://github.com/nicitaacom/23_store/blob/development/app/(auth)/components/ContinueWithButton.tsx

then 

42 line - https://github.com/nicitaacom/23_store/blob/development/app/(auth)/auth/callback/route.ts

then

17 line - https://github.com/nicitaacom/23_store/blob/development/app/(auth)/auth/client/page.tsx

And it doesn't work

Please give me some advice how do I redirect user to production url when I in production and to development url (localhost:3000) when I in development?

**Comment by GaryAustin1** [comment]
The only way is to use redirectTo on the client calls and have that redirectTo full URL switch between the proper URLs to return to.

**Reply by GaryAustin1** [reply]
I see you are attempting that.   If you change your siteUrl around and you get redirected to that all the time then that means the redirectTo URL you are providing is not matching the full URL you have in the dashboard.

EDIT:
Also see this: https://github.com/orgs/supabase/discussions/6832

**Reply by nicitaacom** [reply]
And how can I fix that?

>The only way is to use redirectTo on the client calls

What do you mean by that?
I already have proper URL - location.origin?
https://youtu.be/-7K6DRWfEGM?t=229

here he redirect user to `location.origin/auth/callback`
https://youtu.be/-7K6DRWfEGM?t=237
I do the same:
18 line - https://github.com/nicitaacom/23_store/blob/development/app/(auth)/components/ContinueWithButton.tsx

> and have that redirectTo full URL switch between the proper URLs to return to.

here I as he redirect user to requestUrl (I just added query params to save it in localstorage with zustand)
42 line - https://github.com/nicitaacom/23_store/blob/development/app/(auth)/auth/callback/route.ts
https://youtu.be/-7K6DRWfEGM?t=229

**Comment by simonhackler** [comment]
You probably don't want to develop on your production environment. To setup local development take a look at the following link
https://supabase.com/docs/guides/cli/local-development

**Reply by nicitaacom** [reply]
A lot of stuff to do just to fix this little problem
My question is another - is how to redirect user to localhost:3000 in development and to production url in production

In this guide described about local development but I don't use this guide set up like `supabase init` or `supabase start` or `psql 'postgresql://postgres:postgres@localhost:54322/postgres'` and creating sql without UI with coding in .sql file (its not what I want) - so this guide related to similar things related to `local development` but not to `auth redirect to origin url`

**Comment by silentworks** [comment]
In your screenshot you have a path of `/api/auth/callback` while in your project the `callback` route is located at `/auth/callback`, while you are also pointing in line 18 of your https://github.com/nicitaacom/23_store/blob/development/app/(auth)/components/ContinueWithButton.tsx component to `/auth/callback/` (note the extra backslash at the end). You need to correct two of these things to get this to work correctly.

**Reply by nicitaacom** [reply]
> In your screenshot you have a path of /api/auth/callback while in your project the callback route is located at /auth/callback

I changed it to this (as you can see in screenshot_2 I still get redirect to localhost:3000 in production)
You may make sure about it - https://23-store.vercel.app/
`schreenshot_1`
![image](https://github.com/supabase/supabase/assets/39565703/27c9026f-a306-49b6-9f5e-3ee1bc3b1fa6)
`schreenshot_2`
![image](https://github.com/supabase/supabase/assets/39565703/cf0356d5-e241-4d6a-a928-7ab3462824f6)

**Comment by nicitaacom** [comment, marked as answer]
## Answered by @silentworks 

> In your screenshot you have a path of /api/auth/callback while in your project the callback route is located at /auth/callback, while you are also pointing in line 18 of your https://github.com/nicitaacom/23_store/blob/development/app/(auth)/components/ContinueWithButton.tsx component to /auth/callback/ (note the extra backslash at the end). You need to correct two of these things to get this to work correctly.



## Solution

I needed to add / after url in - `https://supabase.com/dashboard/project/[projectId] - authentication - URL configuration - Redirect URLs` 

`screenshot_3`
![image](https://github.com/supabase/supabase/assets/39565703/a4fa9d11-d938-4507-8376-a1a60fc3cc58)
`screenshot_4`
![image](https://github.com/supabase/supabase/assets/39565703/79a52b50-4369-4293-9b83-873892beefaa)
